modules/extension/wms/src/main/java/org/geotools/ows/wms/map/WMSCoverageReader.java

```diff
@@ -133,6 +133,32 @@ public WMSCoverageReader(WebMapServer wms, Layer layer, String style) {
         }
     }
 
+    public WMSCoverageReader(WebMapServer wms, Layer layer, String style, String preferredFormat) {
+        this.wms = wms;
+
+        // init the reader
+        addLayer(layer, style);
+
+        // best guess at the format with a preference for PNG (since it's normally transparent)
+        List<String> formats = wms.getCapabilities().getRequest().getGetMap().getFormats();
```

Comment:
The comment looks out of context. You are basically getting the supported formats form the remote and then cheeking that the pass by format is supported.

Reply:
fixed it